### ISSUE TITLE
Disabling linker crashes application on startup on iPad

### ISSUE THREAD

**Issue by impworks** [MonoGame/MonoGame#7928]
**Steps to reproduce:**

1. Create a blank application using the MonoGame iOS template
2. Open project properties, set `iOS / Build / Linker behavior` to `Don't link`
3. Run the app on iPad

**Expected result:**
Application starts

**Actual result:**
Application crashes immediately at startup, the debugger hangs and needs to be manually terminated

Surprisingly, it only reproduces on the iPad: running the same app on iPhone with linker disabled works fine.
I cannot avoid setting this property because otherwise [SpriteFonts do not work at all](https://github.com/MonoGame/MonoGame/issues/7868).

Minimal repro code:
https://github.com/impworks/monogame-ios-linker-bug

#### What version of MonoGame does the bug occur on:
- MonoGame 3.8.1.303

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:
- iOS

P.S. This is the fourth (!) iOS-related showstopper bug in a row that I'm running into in a single week:
1. [SpriteFonts do not work](https://github.com/MonoGame/MonoGame/issues/7868)
2. [Content is missing](https://github.com/MonoGame/MonoGame/issues/7897)
3. [Music doesn't play](https://github.com/MonoGame/MonoGame/issues/7197)

**Comment by mrhelmut** [comment]
This doesn't sound like a MonoGame bug but like a Xamarin.iOS bug. Also, "don't link" is not intended to be used for production. If your app needs it, it's very much likely an issue with Xamarin.iOS or a misunderstanding of the linking process in combination with the use of reflection.

**Comment by damian-666** [comment]
As.mentioned sounds definitely trimming or "linking" out your content..u
have to  tinker around  and shut off trimming.. also reflection and dynamic
loading was indeed  problem for me too.  for my plugins code I had so call
at least one method in the code just so that it would not trim out the
plugins.modules in release builds..  it thinks is not calledb/c  there's
not references ,  in the compiled core code so it's trimmed out to reduct
paackage size  .it's not a mg issue ...the package , deploy stuff can't
always figure out what is loaded at runtime..so there's been suggestions
how to link with out trimming an  such in the project files refernces..  I
would look at a those under migration guide, disAable all  trimming or
supposed uncallled code, understand private assets and similar attributes..
then maybe  USB debug it.  and then look at  special tooling around
trimming and "linking" special to ios. It's flaky..because now it works for
me on Android even after  I cleaned out  my project hacks and workarounds
to create a comile time reference......I use vs 2022 preview and have got 5
updates since mg3.8.1  And some issues just go away.. as msfts fix stuff,
monthly on this version

**Comment by damian-666** [comment]
Aren't song broken in iOS? Maybe the  #7197 ?   someoen convert the song to sound effect as a workaround , it's bloated usig Wave file  but works.   Mp3 content  is supposed to work but I haven't figure it out.. I resort to wav file, for now .Also then  u can mix   several songs.  or try another audio module..  the xna api  sound is not so great.on gl SDL platforms,  I'm getting the impression..but in dx its ok..for me so far...

**Comment by impworks** [comment]
@mrhelmut thanks for your reply!

My application doesn't need any reflection by itself, it's only used as a workaround for the content reader loading exception. 

I believe it would be super helpful to a lot of newcomers (such as myself) if you could tweak the default Monogame iOS template to do the linking the "right way" out of the box, or provide a detailed guide how to do it manually (at least for the sample project that does nothing fancy), especially if not doing so triggers a show-stopper bug somewhere down the stack.

**Comment by mrhelmut** [comment]
Yes, the reflection used in the content readers and the AOT implications should be better documented, which is why I'd like to keep this issue and #7868 opened (which are actually the same issue) until we figured a proper documentation and template configuration.
